modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceConfigurationManager.java

```diff
@@ -133,6 +138,38 @@ public ServiceOrder changeServiceConfiguration(
         }
     }
 
+    private void validateAllConfigChangeOrdersCompleted(
+            ServiceDeploymentEntity serviceDeploymentEntity) {
+        List<ServiceOrderEntity> serviceOrders = serviceDeploymentEntity.getServiceOrders();
+        if (CollectionUtils.isEmpty(serviceOrders)) {
+            return;
+        }
+        List<ServiceOrderEntity> configChangeOrders =
+                serviceOrders.stream()
+                        .filter(
+                                serviceOrder ->
+                                        serviceOrder.getTaskType()
+                                                == ServiceOrderType.CONFIG_CHANGE)
+                        .toList();
+        if (!CollectionUtils.isEmpty(configChangeOrders)) {
+            boolean isAllConfigChangeOrdersCompleted =
+                    configChangeOrders.stream()
+                            .allMatch(
+                                    order ->
+                                            order.getOrderStatus() == OrderStatus.SUCCESSFUL
+                                                    || order.getOrderStatus()
+                                                            == OrderStatus.FAILED);
+
+            if (isAllConfigChangeOrdersCompleted) {
+                throw new ServiceOrderNotFound(
```

Comment:
this exception is wrong. We must return - 'There is already a pending config change order. Order ID - xxxx'
We must also need a correct exception type here.